startupmanager/src/main/java/org/apache/sling/launchpad/startupmanager/StartUpHandlerImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.launchpad.startupmanager;
+
+import org.apache.sling.launchpad.api.StartupHandler;
+import org.apache.sling.launchpad.api.StartupMode;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.ServiceFactory;
+
+public class StartUpHandlerImpl implements StartupHandler
+{
+    private final Bundle framework;
+
+    public StartUpHandlerImpl(Bundle bundle)
+    {
+        framework = bundle;
+    }
+
+    @Override
+    public StartupMode getMode()
+    {
+        return StartupMode.RESTART;
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return framework.getState() == Bundle.ACTIVE;
```

Comment:
Shouldn't we wait until the whole framework is started, like in here:
https://github.com/apache/sling-whiteboard/blob/097012ca7db585044aaafe09875b2f5f292ec8b7/startupmanager/src/main/java/org/apache/sling/launchpad/startupmanager/StartupListenerTracker.java#L92-L97

Reply:
@trekawek, I think the two are equivalent. If the system bundle is ACTIVE it will have send the FrameworkEvent.STARTED as it reached the target startlevel.

Reply:
I didn't notice that it's bundle "0". Thanks for the explanation!